x/ccv/provider/keeper/grpc_query.go

```diff
@@ -93,6 +125,19 @@ func (k Keeper) GetConsumerChain(ctx sdk.Context, consumerId string) (types.Chai
 		strDenylist[i] = addr.String()
 	}
 
+	phase, ok := k.GetConsumerPhase(ctx, consumerId)
+	if !ok {
+		return types.Chain{}, fmt.Errorf("cannot find phase for consumer (%s)", consumerId)
+	}
+
+	metadata, err := k.GetConsumerMetadata(ctx, consumerId)
+	if err != nil {
+		return types.Chain{}, fmt.Errorf("cannot get metadata for consumer (%s): %w", consumerId, err)
+	}
+
+	clientID, _ := k.GetConsumerClientId(ctx, consumerId)
+	topN := k.GetTopN(ctx, consumerId)
+
 	return types.Chain{
 		ChainId:            consumerId,
```

Comment:
This shouldn't be `consumerId` here but that's on me. We should have another field in `Chain` that contains the `consumerId`. And `chainId` should be populated from `GetConsumerChainId`.

I can add a note to fix it after you PR goes through.

Reply:
No problem. Updated to use `k.GetConsumerChainId(ctx, consumerId)` instead. Does it sound right to you?